src/CornerstoneViewport/CornerstoneViewport.js

```diff
@@ -281,17 +278,14 @@ class CornerstoneViewport extends Component {
       cornerstoneTools.stopClip(this.element);
     }
 
+    // ~~ OVERLAY
+    if (isOverlayVisible !== prevIsOverlayVisible)
+      updatedState.isOverlayVisible = isOverlayVisible
+
     // ~~ STATE: Update aggregated state changes
     if (Object.keys(updatedState).length > 0) {
       this.setState(updatedState);
     }
-
-    if (isOverlayVisible !== prevIsOverlayVisible)
-      this.setState({ isOverlayVisible });
-
-    // Rebind event listeners when props change to handle functional component method
-    this._bindExternalEventListeners(true);
-    this._bindExternalEventListeners();
```

Comment:
Huh. I guess none of these use props when binding. I'm not sure why we did unbind/rebind